### DISCUSSION TITLE
add min_dist_bad_bin argument for calculate_insulation_score

### DISCUSSION THREAD

**Discussion by gdolsten** [open2c/cooltools#321] [Ideas]
Old issue title:  Insulation score does not ignore "NaN" regions; additionally, can diamond insulation be normalized to surrounding region?

Hi, there are two separate concerns here:
1. The insulation score does not seem to ignore regions that are NaN in the coolfile, leading to discrepancies
2. The diamond insulation score should be normalized to the regions proximal to the diamond, so that you are measuring "insulation compared to background". Is there some program which does this?

**Comment by gfudenberg** [comment]
Thanks for the note & sorry for the slow reply.

It appears we pass:
-    min_frac_valid_pixels=0.66,
-    min_dist_bad_bin=0,
to `find_boundaries`, but should add this as an argument with sensible defaults to the `calculate_insulation_score`. (The current version of calculate_insulation_score assumed the user would use the returned table to filter insulation however they wish for downstream analysis).

The CLI should be updated to pass these to `calculate_insulation_score` as well.

We should decide, however, if we set min_dist_bad_bin=1 by default (it may be more interpretable) cc @golobor

**Comment by gfudenberg** [comment]
is this addressed in new user-facing `insulation` function ? cc @Phlya @agalitsyna

**Comment by agalitsyna** [comment]
insulation calculation of cooltools can now ignore NaNs and neighboring regions, controlled by min_dist_bad_bin param (introduced with pre-0.5.0 updates here https://github.com/open2c/cooltools/commit/98b11db82f799f120bc9a25026d92b1683297aba). 

@gfudenberg , what do you think about the normalization of insulation to the background?

**Comment by gfudenberg** [comment]
Do you mean the `norm_by_median` argument in insul_diamond that is not
passed along to the user-facing function ?

**Comment by agalitsyna** [comment]
Ah, okay, I had another explanation in mind. There is another widespread type of insulation normalization where the IS track is divided not by the whole chromosome average but by the moving centered average. This is not a traditional insulation score definition, and I am not sure who first implemented it. Some examples of "moving average insulation": [[1]](https://www.sciencedirect.com/science/article/pii/S0092867417303434#sec4) and [[2]](https://vaquerizaslab.github.io/fanc/fanc-executable/fanc-analyse-hic/domains.html). cooltools has the traditional insulation right now, following the original [Crane et al. 2016 work](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4498965/#S1title)

**Comment by gfudenberg** [comment]
got it-- sure I'd think a local option could be reasonable to add. other thoughts? @agalitsyna  @golobor @Phlya

**Comment by golobor** [comment]
I am not sure at the moment - what's the motivation for this local normalization? I'd suggest that we do some studies before adding such a non-trivial feature to the main code.

**Comment by Phlya** [comment]
Oh missed this. Yes, I agree with Anton - is there a reason to do this? It shouldn't really affect positions of dips of insulation anyway... And whether it would improve anything is not at all obvious to me. Is there a use case where it works better?

**Comment by gfudenberg** [comment]
Should we close this / move to discussion ?

**Reply by agalitsyna** [reply]
Thanks, @gfudenberg! Let's keep it as a discussion point. We still received no specific justification for the addition of window average normalization in insulation. Probably the only thing in favor of insulation modification I found in the literature is this example, although yet not exactly about this type of insulation normalization: https://academic.oup.com/view-large/figure/126093371/gky789fig1.jpg [[1]](https://academic.oup.com/nar/article/46/21/11239/5090284)
